tableaudocumentapi/field.py

```diff
@@ -117,51 +136,199 @@ def id(self):
         return self._id
 
     @property
+    def xml(self):
+        """ XML representation of the field. """
+        return self._xml
+
+    ########################################
+    # Attribute getters and setters
+    ########################################
+
+    @property
     def caption(self):
         """ Name of the field as displayed in Tableau unless an aliases is defined """
         return self._caption
 
+    @caption.setter
+    def caption(self, caption):
+        """ Set the caption of a field
+
+            Args:
+                caption:  New caption. String.
+
+            Returns:
+                Nothing.
+        """
+        self._caption = caption
+        self._xml.set('caption', caption)
+
     @property
     def alias(self):
         """ Name of the field as displayed in Tableau if the default name isn't wanted """
         return self._alias
 
+    @alias.setter
+    def alias(self, alias):
+        """ Set the alias of a field
+
+            Args:
+                alias:  New alias. String.
+
+            Returns:
+                Nothing.
+        """
+        self._alias = alias
+        self._xml.set('alias', alias)
+
     @property
     def datatype(self):
         """ Type of the field within Tableau (string, integer, etc) """
         return self._datatype
 
+    @datatype.setter
+    @argument_is_one_of('string', 'integer', 'date', 'boolean')
+    def datatype(self, datatype):
+        """ Set the datatype of a field
+
+            Args:
+                datatype:  New datatype. String.
+
+            Returns:
+                Nothing.
+        """
+        self._datatype = datatype
+        self._xml.set('datatype', datatype)
+
     @property
     def role(self):
         """ Dimension or Measure """
         return self._role
 
+    @role.setter
+    @argument_is_one_of('dimension', 'measure')
+    def role(self, role):
+        """ Set the role of a field
+
+            Args:
+                role:  New role. String.
+
+            Returns:
+                Nothing.
+        """
+        self._role = role
+        self._xml.set('role', role)
+
+    @property
+    def type(self):
```

Comment:
I've removed all 'free floating' variables that are named 'type'.
I didn't change the properties that are named 'type', since they correspond to a XML attribute named 'type' and changing it will throw off anyone using the API. The way it's working, using type() should not be affected by this.